lib/dnsmsg-parser/src/dns_message_parser.rs

```diff
@@ -338,290 +341,182 @@ impl DnsMessageParser {
         rdata: &NULL,
     ) -> DnsParserResult<(Option<String>, Option<Vec<u8>>)> {
         match code {
-            dns_message::RTYPE_MB => match rdata.anything() {
-                Some(raw_rdata) => {
-                    let mut decoder = self.get_rdata_decoder_with_raw_message(raw_rdata);
-                    let madname = parse_domain_name(&mut decoder)?;
-                    Ok((Some(madname.to_string()), None))
-                }
-                None => Err(DnsMessageParserError::SimpleError {
-                    cause: String::from("Empty MB rdata"),
-                }),
-            },
+            dns_message::RTYPE_MB => {
+                let mut decoder = self.get_rdata_decoder_with_raw_message(rdata.anything());
```

Comment:
This was the bulk of the changes here. `RData#anything` was changed to not return an `Option`. Instead `Record#data` returns an `Option<RData>` (you can see the corresponding change above around 152).

https://github.com/bluejekyll/trust-dns/pull/1626

It's not _really_ clear to me when there would be no record data, but the end result should be preserved that we just bubble up an error.